src/packages/web-worker/command-handlers/host-call.ts

```diff
@@ -165,6 +165,12 @@ const hostCall = async ({
     await jamHostCall.execute(getGasCounter(pvm), getRegisters(pvm), getMemory(pvm));
 
     return { hostCallIdentifier, storage, status: CommandStatus.SUCCESS };
+  } else if (hostCallIdentifier === HostCallIdentifiers.GAS) {
```

Comment:
https://github.com/FluffyLabs/typeberry/blob/main/packages/pvm/host-calls/host-calls-manager.ts#L10

Take a look at this code it's doing exactly the thing that you are doing here with a series of if-elses and handles a missing host call by returning some info back to the PVM.
You might consider using this instead.